rrule/src/parser/regex.rs

```diff
@@ -50,7 +44,15 @@ impl ParsedDateString {
     /// Parses a date string with format `YYYYMMDD(THHMMSSZ)` where the part in parentheses
     /// is optional. It returns [`ParsedDateString`].
     pub(crate) fn from_ical_datetime(val: &str) -> Result<Self, ParseError> {
+        static DATESTR_RE: OnceLock<Regex> = OnceLock::new();
+
         let captures = DATESTR_RE
+            .get_or_init(|| {
+                Regex::new(
+                    r"(?m)^([0-9]{4})([0-9]{2})([0-9]{2})(T([0-9]{2})([0-9]{2})([0-9]{2})(Z?))?$",
+                )
+                .expect("DATESTR_RE regex failed")
```

Comment:
Message for `expect` should describe the expectation, not the opposite.

```suggestion
                .expect("DATESTR_RE must compile")
```